sources/packages/backend/apps/api/src/route-controllers/education-program-offering/education-program-offering.institutions.controller.ts

```diff
@@ -180,16 +243,58 @@ export class EducationProgramOfferingInstitutionsController extends BaseControll
         userToken.userId,
       );
     } catch (error: unknown) {
-      if (
-        error instanceof CustomNamedError &&
-        error.name === OFFERING_VALIDATION_CRITICAL_ERROR
-      ) {
-        throw new BadRequestException(error.objectInfo, error.message);
+      if (error instanceof CustomNamedError) {
+        switch (error.name) {
+          case OFFERING_INVALID_OPERATION_IN_THE_CURRENT_STATE:
+            throw new UnprocessableEntityException(error.message);
+          case OFFERING_VALIDATION_CRITICAL_ERROR:
+            throw new BadRequestException(error.objectInfo, error.message);
+        }
       }
       throw error;
     }
   }
 
+  /**
+   * Updates offering basic information that can be freely changed
+   * without affecting the assessment.
+   * @param payload offering data to be updated.
+   * @param locationId offering location.
+   * @param programId offering program.
+   * @param offeringId offering to be modified.
+   */
+  @HasLocationAccess("locationId")
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Either offering for the program and location is not found or the offering is not in the appropriate status to be updated.",
+  })
+  @Patch(
+    "location/:locationId/education-program/:programId/offering/:offeringId/basic",
+  )
+  async updateProgramOfferingBasicInformation(
+    @UserToken() userToken: IInstitutionUserToken,
+    @Param("locationId", ParseIntPipe) locationId: number,
+    @Param("programId", ParseIntPipe) programId: number,
+    @Param("offeringId", ParseIntPipe) offeringId: number,
+    @Body() payload: EducationProgramOfferingBasicDataAPIInDTO,
+  ): Promise<void> {
+    const offering = await this.programOfferingService.getProgramOffering(
+      locationId,
+      programId,
+      offeringId,
+    );
+    if (!offering) {
+      throw new UnprocessableEntityException(
+        "Either offering for the program and location is not found or the offering is not in the appropriate status to be updated.",
```

Comment:
why are we not passing the 4th parameter to check if the offering is in appropriate status to be updated(even though it is basic data) ? 
![image](https://user-images.githubusercontent.com/54600590/200693969-57ddaa2c-6954-47cb-9c7a-5797e9b88110.png)

![image](https://user-images.githubusercontent.com/54600590/200693940-f6346cea-e67e-4424-8ea4-ed5df9c62caf.png)

Reply:
Good point, I will add it.

Reply:
Changed.